base/src/Data/Macaw/Analysis/RegisterUse.hs

```diff
@@ -352,191 +1315,316 @@ requiredAssignDeps :: OrdF (ArchReg arch)
                    -> DependencySet (ArchReg arch) ids
                    -> RegisterUseM arch ids ()
 requiredAssignDeps aid deps = do
-  addDeps $ deps
+  addDeps deps
   assignmentCache %= Map.insert (Some aid) mempty
 
+popAccessInfo :: RegisterUseM arch ids (MemAccessInfo arch ids)
+popAccessInfo = do
+  s <- get
+  case pendingMemAccesses s of
+    [] -> error "popAccessInfo invalid"
+    (i:r) -> do
+      put $! s { pendingMemAccesses = r }
+      pure i
+
+demandReadMem :: (MemWidth (ArchAddrWidth arch), OrdF (ArchReg arch))
+              => AssignId ids tp
+              -> Value arch ids (BVType (ArchAddrWidth arch))
+              -> MemRepr tp
+              -> RegisterUseM arch ids ()
+demandReadMem aid addr _repr = do
+  accessInfo <- popAccessInfo
+  case accessInfo of
+    NotFrameAccess -> do
+      -- Mark that this value depends on both aid and any
+      -- dependencies needed to compute the address.
+      cns <- gets blockUsageStartConstraints
+      cache <- gets assignDeps
+      let deps = assignSet aid <> valueDeps cns cache addr
+      requiredAssignDeps aid deps
+    FrameReadInitAccess _o d -> do
+      let deps = assignSet aid <> domainDeps d
+      assignmentCache %= Map.insert (Some aid) deps
+    FrameReadWriteAccess _o writeIdx -> do
+      -- Mark that this value depends on aid and any
+      -- dependencies needed to compute the value stored at o.
+      m <- gets blockWriteDependencies
+      let deps = Map.findWithDefault (error "Could not find write index.") writeIdx m
+      let allDeps = addWriteDep writeIdx (assignSet aid <> deps)
+      assignmentCache %= Map.insert (Some aid) allDeps
+    FrameReadOverlapAccess _ -> do
+      assignmentCache %= Map.insert (Some aid) (assignSet aid)
+    FrameWriteAccess{} ->
+      error "Expected read access."
+    FrameCondWriteAccess{} ->
+      error "Expected read access."
+    FrameCondWriteOverlapAccess _ ->
+      error "Expected read access."
+
+demandCondReadMem :: (MemWidth (ArchAddrWidth arch), OrdF (ArchReg arch))
+                  => AssignId ids tp
+                  -> Value arch ids BoolType
+                  -> Value arch ids (BVType (ArchAddrWidth arch))
+                  -> MemRepr tp
+                  -> Value arch ids tp
+                  -> RegisterUseM arch ids ()
+demandCondReadMem aid cond addr _repr val = do
+  accessInfo <- popAccessInfo
+  case accessInfo of
+    NotFrameAccess -> do
+      cns   <- gets blockUsageStartConstraints
+      cache <- gets assignDeps
+      let deps = mconcat
+                    [ assignSet aid
+                    , valueDeps cns cache cond
+                    , valueDeps cns cache addr
+                    , valueDeps cns cache val
+                    ]
+      requiredAssignDeps aid deps
+      addDeps $ assignSet aid
```

Comment:
Why does this (`CondReadMem`) case get this call to `addDeps` while the analogous case in the `ReadMem` handler above does not?